### PR TITLE
Implement basic multi-region support

### PR DESCRIPTION
**Pull request by jonathanvdc**
This PR adds support for sequential and parallel multi-region simulations. Ironically, about half of this PR is just a refactor of the Stride source code. Stride's liberal use of a global log had to be eliminated to make multi-threaded simulation logs sensible, and I had to take `run_stride.cpp` apart before putting it together again in order to support multi-region simulations.

I also set  `AlignAfterOpenBracket: AlwaysBreak` in the `.clang-format` file. I hope that's not too controversial.

### CODE REVIEW

File: src/main/cpp/pop/PopulationBuilder.h [hunk 1]
```diff
@@ -43,12 +44,12 @@ class PopulationBuilder
...
+	 * @param log             The contact log.
 	 * @return                Pointer to the initialized population.
 	 */
-	static std::shared_ptr<Population> Build(
-	        const SingleSimulationConfig& config,
-	        const boost::property_tree::ptree& pt_disease,
-	        util::Random& rng);
+	static std::shared_ptr<Population> Build(const SingleSimulationConfig& config,
+						 const boost::property_tree::ptree& pt_disease, util::Random& rng,
+						 const std::shared_ptr<spdlog::logger>& log);
```
Comment (lynn) [inline src/main/cpp/pop/PopulationBuilder.h]: This file probably needs re-`clang-format`-ing?

File: src/main/cpp/sim/run_stride.cpp [hunk 2]
```diff
@@ -53,135 +58,205 @@ using namespace boost::property_tree;
...
+	// Create simulator.
+	// -----------------------------------------------------------------------------------------
+	Stopwatch<> total_clock("total_clock", true);
+	multiregion::ParallelSimulationManager<StrideSimulatorResult, size_t> sim_manager{num_threads};
+
+	// Build all the simulations.
+	size_t config_index = 0;
+	std::vector<std::tuple<
+	    std::string, std::string, SingleSimulationConfig,
+	    std::shared_ptr<multiregion::SimulationTask<StrideSimulatorResult>>>>
```
Comment (lynn) [inline src/main/cpp/sim/run_stride.cpp]: This `std::tuple` deserves to be its own struct with descriptive record names, I feel.

**Review by lynn — CHANGES_REQUESTED** [review]
This code looks great! Other than the small changes I’ve mentioned, I’m also not entirely happy with the `boost::any` stuff, but there’s probably no way around it. \*shakes fist\* *Bjarne!!!*

File: src/main/cpp/pop/PopulationBuilder.h (hunk 1 above)
Comment (jonathanvdc) [reply src/main/cpp/pop/PopulationBuilder.h]: Job's done

File: src/main/cpp/sim/run_stride.cpp (hunk 2 above)
Comment (jonathanvdc) [reply src/main/cpp/sim/run_stride.cpp]: All right. I've rewritten it as a local `struct` definition.

**Review by lynn — APPROVED** [review]

**Pull request merged**